ea-2022-04-ndvi-automation.py

```diff
@@ -0,0 +1,645 @@
+
+path = os.path.normpath(os.getcwd())
+student_wd_parts = path.split(os.sep)
+
+if student_wd_parts[-2:] == ['earth-analytics', 'data']:
+    print("\u2705 Great - it looks like your working directory is set correctly to ~/earth-analytics/data")
+else:
+    print("\u274C Oops, the autograder will not run unless your working directory is set to earth-analytics/data")
+
+
+# In[4]:
+
+
+# Site data
+path = os.path.join('ndvi-automation', 'sites')
+
+all_sites_dirs = glob(path + "/*/")
+for site_dirs in all_sites_dirs:
+    print(site_dirs)
+
+site_name = os.path.basename(os.path.normpath(all_sites_dirs[1]))
+site_name
+
+
+# In[5]:
+
+
+# Shapefile data
+vector_dir = os.path.join(all_sites_dirs[1], "vector")
+
+# Create boundary
+site_boundary_path = os.path.join(vector_dir, site_name + "-crop.shp")
+crop_bound = gpd.read_file(site_boundary_path)
+
+# Test boundary plot
+crop_bound.plot()
+plt.show()
+
+
+# In[6]:
+
+
+# Test cell
+landsat_dir = os.path.join(site_dirs,
+                           "landsat-crop")
+# This is the crop folder containing all of the .tif files
+landsat_dirs = sorted(glob(os.path.join(landsat_dir, "LC08*")))
+landsat_dirs
+
+
+# # Figure 1: Plot 1 - Mean NDVI For Each Site Across the Year (50 points)
+# 
+# Create a plot of the mean normalized difference vegetation index (NDVI) for the two different sites in the United States across the year: 
+# 
+# * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object.
+# * Each site should be identified with a different color in the plot and legend.
+# * The final plot **data should be cleaned to remove the influence of clouds**.
+# * Be sure to include appropriate title and axes labels.
+# 
+# Add additional cells as needed for processing data (e.g. defining functions, etc), but be sure to:
+# * follow the instructions in the code cells that have been provided to ensure that you are able to use the sanity check tests that are provided. 
+# * include only the plot code in the cell identified for the final plot code below
+
+# ## Task 1: 
+# 
+# In the cell below, create a single dataframe containing MEAN NDVI, the site name, 
+# and the date of the data for the HARV site 
+# scene `HARV/landsat-crop/LC080130302017031701T1-SC20181023151837`.  The column names for the  final
+# DataFrame should be`mean_ndvi`, and `site`, and the data should be **indexed on the date**. 
+# 
+# Use the functions that we reviewed in class (or create your own versions of them) to implement your code
+# 
+# ### In the Cell below Place  All Functions Needed to Run this Notebook (20 points)
+
+# In[7]:
+
+
+# In this cell place all of the functions needed to run your notebook
+# You will be graded here on function application, docstrings, efficiency so ensure
+# All functions are placed here!
+# Function A
+def open_clean_bands(band_path,
+                     valid_range=None):
+    """Open and mask a single Landsat band using a pixel_qa layer.
```

Comment:
You seem to be masking using the nodata values here, not with the pixel_qa layer. I don't see anywhere that you're using that

Reply:
I believe this is just a misleading docstring. Edited the docstring to correctly describe the functionality.